torch/csrc/autograd/engine.cpp

```diff
@@ -451,11 +451,14 @@ static void validate_outputs(const edge_list& edges, variable_list& grads, const
       }
       grads[i] = at::sum_to(std::move(grads[i]), metadata.shape());
     }
+    if (metadata.type().scalarType() != grads[i].type().scalarType()) {
```

Comment:
can we either throw a check or assert that the grad type is floating point?  Our gradchecks would catch this but I think this path will also be hit by user defined functions.

Reply:
I'll add the check, but I'd like to understand the mechanism where udf's reach this. I couldn't run backward on an int tensor. I.e. is there a way I can add a unit test that would hit an assertion here or is it something that would only be reachable if someone added other incorrect code?

Reply:
the only case I can think where this would be hit would be if you have a user defined torch.autograd.Function and return a long tensor.